src/ray/raylet/scheduling/cluster_resource_data.cc

```diff
@@ -164,6 +164,27 @@ NodeResources ResourceMapToNodeResources(
   return node_resources;
 }
 
+float NodeResources::CalculateCriticalResourceUtilization() const {
+  float highest = 0;
+
+  for (const auto &capacity : predefined_resources) {
+    float utilization = 1 - (capacity.available.Double() / capacity.total.Double());
+    if (utilization > highest) {
+      highest = utilization;
+    }
+  }
```

Comment:
Can we have this only for CPU and object_store_memory? It can have unexpected effects on custom resources.

Reply:
Hmmm good point... I can imagine similar situations for CPU and memory though.

How about we do all predefined resources and no custom resources?